data-runtime/src/main/java/io/micronaut/data/runtime/mapper/JsonColumnReader.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.runtime.mapper;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.util.ArgumentUtils;
+import io.micronaut.data.exceptions.DataAccessException;
+import io.micronaut.json.JsonMapper;
+
+import java.io.IOException;
+
+/**
+ * The default JSON column reader implementation.
+ *
+ * @author radovanradic
+ * @since 4.0.0
+ *
+ * @param <RS> the result set type
+ */
+public class JsonColumnReader<RS> {
```

Comment:
Make this an interface + default implementation

Reply:
Changed